hcl/hcl_script.go

```diff
@@ -0,0 +1,148 @@
+	Description *ScriptDescription // Description is a human readable description of a script
+	Jobs        []*ScriptJob       // Job represents the command(s) part of this script
+}
+
+// NewScriptDescription returns a *ScriptDescription encapsulating an ast.Attribute
+func NewScriptDescription(attr ast.Attribute) *ScriptDescription {
+	desc := ScriptDescription(attr)
+	return &desc
+}
+
+// NewScriptCommand returns a *Command encapsulating an ast.Attribute
+func NewScriptCommand(attr ast.Attribute) *Command {
+	cmd := Command(attr)
+	return &cmd
+}
+
+// NewScriptCommands returns *Commands encapsulating an ast.Attribute
+func NewScriptCommands(attr ast.Attribute) *Commands {
+	cmds := Commands(attr)
+	return &cmds
+}
+
+func (p *TerramateParser) parseScriptBlock(block *ast.Block) (*Script, error) {
+	errs := errors.L()
+
+	parsedScript := &Script{
+		Labels: block.Labels,
+	}
+
+	for _, attr := range block.Attributes {
+		switch attr.Name {
+		case "description":
+			parsedScript.Description = NewScriptDescription(attr)
+		default:
+			errs.Append(errors.E(ErrScriptUnrecognizedAttr, attr.NameRange))
+		}
+	}
+
+	for _, nestedBlock := range block.Blocks {
+		switch nestedBlock.Type {
+		case "job":
+			parsedJobBlock, err := validateScriptJobBlock(nestedBlock)
+			if err != nil {
+				errs.Append(err)
+				continue
+			}
+			parsedScript.Jobs = append(parsedScript.Jobs, parsedJobBlock)
+		default:
+			errs.Append(errors.E(ErrScriptUnrecognizedBlock, nestedBlock.TypeRange, nestedBlock.Type))
+
+		}
+	}
+
+	if len(parsedScript.Labels) == 0 {
+		errs.Append(errors.E(ErrScriptNoLabels, block.TypeRange))
+	}
+
+	if len(parsedScript.Jobs) == 0 {
+		errs.Append(errors.E(ErrScriptMissingOrInvalidJob, block.OpenBraceRange))
+	}
+
+	if parsedScript.Description == nil {
+		errs.Append(errors.E(ErrScriptNoDesc, block.OpenBraceRange))
+	}
+
+	if err := errs.AsError(); err != nil {
+		return nil, err
+	}
+
+	return parsedScript, nil
+
+}
+
+func validateScriptJobBlock(block *ast.Block) (*ScriptJob, error) {
+	errs := errors.L()
+
+	parsedScriptJob := &ScriptJob{}
+	for _, attr := range block.Attributes {
+		switch attr.Name {
+		case "command":
+			parsedScriptJob.Command = NewScriptCommand(attr)
+		case "commands":
+			parsedScriptJob.Commands = NewScriptCommands(attr)
+		default:
+			errs.Append(errors.E(ErrScriptUnrecognizedAttr, attr.NameRange, attr.Name))
+
+		}
+	}
+
+	for _, childBlock := range block.Blocks {
+		errs.Append(errors.E(ErrScriptUnrecognizedBlock, childBlock.TypeRange, childBlock.Type))
+	}
+
+	// job.command and job.commands are mutually exclusive
+	if parsedScriptJob.Command != nil && parsedScriptJob.Commands != nil {
+		errs.Append(errors.E(ErrScriptCmdConflict, parsedScriptJob.Command.NameRange))
+		errs.Append(errors.E(ErrScriptCmdConflict, parsedScriptJob.Commands.NameRange))
+	}
+
+	if parsedScriptJob.Command == nil && parsedScriptJob.Commands == nil {
+		errs.Append(errors.E(ErrScriptNoCmds, block.TypeRange))
```

Comment:
```suggestion
		errs.Append(errors.E(ErrScriptNoCmds, block.Range))
```

Reply:
Fixed in https://github.com/terramate-io/terramate/pull/1236/commits/0bd145bc88f507782997876f06b14dc3a983a24c